log_transfer/tasks.py

```diff
@@ -36,67 +46,136 @@ def init():
-    return rs
+    @abstractmethod
+    def mark_as_sent(self) -> None:
+        return NotImplemented
 
-# Used for tests only
-def delete_elastic_index():
-    es=init()
-    if es == None:
-        return
-    es.options(ignore_status=[404]).indices.delete(index=settings.ELASTICSEARCH_APP_AUDIT_LOG_INDEX)
+    def __str__(self) -> str:
+        return str(self.log)
+
+
+class SimpleAuditLogFacade(AuditLogFacade):
+    log: AuditLogEntry
+
+    @cached_property
+    def message(self) -> Dict[str, Any]:
+        message = self.log.message.copy()
+        message["@timestamp"] = (
+            message[settings.DATE_TIME_PARENT_FIELD][settings.DATE_TIME_FIELD]
+            if settings.DATE_TIME_PARENT_FIELD
+            else message[settings.DATE_TIME_FIELD]
+        )
+        return message
+
+    def mark_as_sent(self) -> None:
+        self.log.is_sent = True
+        self.log.save()
+
+
+class DjangoAuditLogFacade(AuditLogFacade):
+    log: "LogEntry"
+
+    @cached_property
+    def message(self) -> Dict[str, Any]:
+        return {
+            "@timestamp": self.log.timestamp,
+            "audit_event": {
+                "actor": self.log.actor.get_full_name() or str(self.log.actor.email),
+                "date_time": self.log.timestamp,
+                "operation": str(self.log.action),
+                "origin": settings.AUDIT_LOG_ORIGIN,
+                "target": self.log.object_repr,
+                "environment": settings.AUDIT_LOG_ENVIRONMENT,
+                "message": self.log.changes,
+            }
+        }
+
+    def mark_as_sent(self) -> None:
+        if self.log.additional_data is None:
+            self.log.additional_data = {}
+        self.log.additional_data["is_sent"] = True
+        self.log.save()
+
+
+def get_unsent_entries() -> List[AuditLogFacade]:
+
+    if settings.AUDIT_LOGGER_TYPE == AuditLoggerType.SINGLE_COLUMN_JSON:
+        return [
+            SimpleAuditLogFacade(log=log)
+            for log in AuditLogEntry.objects.filter(is_sent=False).order_by("created_at")
+        ]
+
+    elif settings.AUDIT_LOGGER_TYPE == AuditLoggerType.DJANGO_AUDITLOG:
+        from auditlog.models import LogEntry
+
+        return [
+            DjangoAuditLogFacade(log=log)
+            for log in LogEntry.objects.filter(
+                ~Q(additional_data__has_key="is_sent")  # support old entries
+                | Q(additional_data__is_sent=False),
+            ).order_by("timestamp")
+        ]
 
+    # Should never happen, but just in case
+    else:
+        raise RuntimeError("Unknown audit logger type set.")
 
-def clear_audit_log_entries(days_to_keep=30):
+
+def clear_audit_log_entries(days_to_keep: int = 30) -> None:
     # Only remove entries older than `X` days
-    sent_entries = AuditLogEntry.objects.filter(
-        is_sent=True, created_at__lte=(timezone.now() - timedelta(days=days_to_keep))
-    )
-    sent_entries.delete()
+
+    if settings.AUDIT_LOGGER_TYPE == AuditLoggerType.SINGLE_COLUMN_JSON:
```

Comment:
Suggest using match-case instead of if sentences for this part now that we use enums. This would improve code readability slightly, however this is only a suggestion for code readability improvement, not mandatory.

Reply:
Oh, sorry, didn't check the python version. Will keep it this way then, especially if it is preferred to do it this way in python.